Error with Chromedriver assuming Chrome has crashed - running headless

Getting the following error when I try to run pytest:

```
E       WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally
E         (unknown error: DevToolsActivePort file doesn't exist)
E         (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
E         (Driver info: chromedriver=2.46.628388 (4a34a70827ac54148e092aafb70504c4ea7ae926),platform=Linux 4.9.0-7-amd64 x86_64)

venv/local/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py:242: WebDriverException
```

ChromeDriver 73.0.3683.20 
Google Chrome 72.0.3626.109 
chromedriver: /usr/bin/chromedriver /usr/local/bin/chromedriver
google-chrome: /usr/bin/google-chrome /usr/share/man/man1/google-chrome.1.gz

Wondering whether this has something to do with me running a headless server, e.g. the only way I can run google-chrome is with --headless

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 Versions follow [Semantic Versioning](https://www.semver.org)
 
+## [2.1.0] - 2019-02-20
+### Added
+- Add `pytest_setup_selenium` hook [#65](https://github.com/T4rk1n/pytest-dash/pull/65)
+
 ## [2.0.0] - 2019-02-12
 ### Added
 - Behavior tests from yaml description. [#54](https://github.com/T4rk1n/pytest-dash/pull/54)

--- a/docs/usage.rst
+++ b/docs/usage.rst
@@ -291,6 +291,9 @@ instead of having to enter it on the command line every time you run a test.
 Hooks
 -----
 
+pytest_add_behaviors
+====================
+
 The scenario event/outcome syntax can be extended with
 the :py:func:`~.new_hooks.pytest_add_behaviors` hook.
 
@@ -317,3 +320,24 @@ the :py:func:`~.new_hooks.pytest_add_behaviors` hook.
 .. seealso::
 
     Lark grammar reference https://lark-parser.readthedocs.io/en/latest/grammar/
+
+pytest_setup_selenium
+=====================
+
+If you need to configure the selenium driver used by the plugin, you can use
+the ``pytest_setup_selenium`` hook.
+
+:Example: ``tests/conftest.py``
+
+.. code-block:: python
+    :caption: Run chrome in headless mode.
+
+    from selenium.webdriver.chrome.options import Options
+
+
+    def pytest_setup_selenium(driver_name):
+        options = Options()
+        options.headless = True
+        return {
+            'chrome_options': options,
+        }

--- a/pytest_dash/__init__.py
+++ b/pytest_dash/__init__.py
@@ -3,4 +3,4 @@ Pytest-dash
 -----------
 Pytest fixtures and helper methods for dash.
 """
-__version__ = '2.0.0'
+__version__ = '2.1.0'

--- a/pytest_dash/application_runners.py
+++ b/pytest_dash/application_runners.py
@@ -30,7 +30,7 @@ def _stop_server():
 def _assert_closed(driver):
     driver.refresh()
     body = driver.find_element_by_css_selector('body').text
-    return 'refused to connect' in body
+    return 'refused to connect' in body or not body
 
 
 def _handle_error(_):

--- a/pytest_dash/new_hooks.py
+++ b/pytest_dash/new_hooks.py
@@ -20,3 +20,15 @@ def pytest_add_behaviors(add_behavior):
     :param add_behavior: Decorator for a behavior handler function.
     :return:
     """
+
+
+def pytest_setup_selenium(driver_name):
+    """
+    Called before the driver is created, return a dictionary to use as kwargs
+    for the driver init.
+
+    :param driver_name: The name of the driver specified by either cli
+        argument or in pytest.ini.
+    :type driver_name: str
+    :return: The dictionary of kwargs to give to the driver constructor.
+    """

--- a/pytest_dash/plugin.py
+++ b/pytest_dash/plugin.py
@@ -112,7 +112,13 @@ class DashPlugin(object):
                     )
                 )
 
-            self._driver = _driver_map.get(self._driver_name)()
+            options = {}
+            hooked_options = self.config.hook.pytest_setup_selenium(
+                driver_name=self._driver_name
+            ) or []
+            for opt in hooked_options:
+                options.update(opt)
+            self._driver = _driver_map.get(self._driver_name)(**options)
 
         return self._driver
 

